saw/SAWScript/REPL/Trie.hs

```diff
@@ -41,6 +41,18 @@ lookupTrie key t@(Node mp _) = case key of
 
   [] -> leaves t
 
+-- | Return all matches with the given prefix. However, if an exact match
+-- exists, return just that match.
+lookupTrieWithExact :: String -> Trie a -> [a]
+lookupTrieWithExact key t@(Node mp mb) = case key of
+
+  c:cs -> case Map.lookup c mp of
+    Just m' -> lookupTrieWithExact cs m'
+    Nothing -> []
+
+  [] | Just b <- mb -> [b]
+  [] | otherwise -> leaves t
```

Comment:
I think this would be more directly expressed with a case expression rather than with guards:

```suggestion
  [] -> case mb of
    Just b  -> [b]
    Nothing -> leaves t
```

Reply:
agreed, fixed in be94c5fc1511f5aebdc2ee25d18cf5bbff2a86dd